### PR TITLE
Implement make_unique

### PR DESCRIPTION
**Pull request by gassmoeller**
This will be useful in a follow up PR, and also in other places.

### CODE REVIEW

File: include/aspect/compat.h [hunk 1]
```diff
@@ -146,5 +146,25 @@ namespace aspect
 }
 #endif
 
+#if !DEAL_II_VERSION_GTE(9,0,0)
```
Comment (tjhei) [inline include/aspect/compat.h]: I think you need to ``#include <std_cxx14/memory.h>`` in an #else block.

**Comment by tjhei** [comment]
Can you also leave a comment in that function that this is just a hack and doesn't allow for constructors with parameters? Otherwise people might be surprised later.

**Comment by gassmoeller** [comment]
Done

**Comment by bangerth** [comment]
What a shame that we can't use C++11 in ASPECT yet :-(

**Pull request merged**